StRoot/StFcsWaveformFitMaker/StFcsWaveformFitMaker.cxx

```diff
@@ -620,6 +615,45 @@ TGraphAsymmErrors* StFcsWaveformFitMaker::makeTGraphAsymmErrors(StFcsHit* hit){
     return gae;
 }
 
+
+float StFcsWaveformFitMaker::AnaPed( TGraphAsymmErrors* g, float& ped, float& pedstd )
+{
+  int n = g->GetN();
+  double *t = g->GetX();
+  double *a = g->GetY();    
+  int p=0;
+  double sumsq = 0;//For variance
+  for(int i=0; i<n; i++){
+    int tb1=t[i];
+    if(mPedMin<=tb1 && tb1<=mPedMax){    
+      p+=a[i];
+      sumsq += a[i]*a[i];
+    }
+  }
+  ped = float(p)/(mPedMax-mPedMin+1.0);
+  pedstd = sqrt( ( sumsq-((double(p)*double(p))/(mPedMax-mPedMin+1.0)) )/(mPedMax-mPedMin) );//Variance/StdDev using naive algorithm from https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance
+  return ped;
```

Comment:
Why return the same value if you modify the function argument?

Reply:
This has to do with the way the code was setup before. I didn't want to overhaul too many things and preserve the overall logic in Make. I would have to talk to @akioogawa before making such changes because it could break the way he has things set up and will require more time and testing to ensure that everything still works as intended.